prototype/cmd/client-with-pre-p2p/main.go

```diff
@@ -0,0 +1,176 @@
+package main
```

Comment:
Rather than copy-pasting main.go, maybe we can just create config internally. The goal of it was for both of them to have the same interface.

Reply:
This was a mindless copy-paste and move-around to simply add p2p and keep everything that was working, working.

I am unclear about what you said about creating the config internally. We can make a separate PR for the client optimizations and refactors.